protocol/smart-batching.md

```diff
@@ -44,6 +44,8 @@ blob data.
 
 Modify the L1 derivation stage of the pipeline (`calldata_source` and `blob_data_source`) to fetch receipts for each L1 block they process, and skip any transaction where the `status` field of the receipt is 0 (reverted transactions).
 
+To support existing chains opting into this feature, we also make it possible for chains to update their inbox address, by adding an owner-only `setBatchInbox` method to the `SystemConfig.sol` contract. Chains can then use the existing mechanisms for updating the `SystemConfig` to change their batch inbox address to that of a smart contract.
```

Comment:
This will also need to take into account passing the information into the L2 system, so there will likely need to be an [UpdateType](https://github.com/ethereum-optimism/optimism/blob/aec52922e9aa194b5288f2cd466b5799a3d402e5/packages/contracts-bedrock/src/L1/SystemConfig.sol#L28) and the [L1 attributes tx calldata](https://github.com/ethereum-optimism/optimism/blob/aec52922e9aa194b5288f2cd466b5799a3d402e5/packages/contracts-bedrock/src/L2/L1Block.sol#L195) will need to include the batch inbox address. Its not ideal to keep bloating the size of the L1 attributes tx but its a clear way to continue to add config and would result in a larger diff to the proof system if we didn't include it in the calldata